xbee_android_library/src/main/java/com/digi/xbee/api/android/connection/android/AndroidXBeeInterface.java

```diff
@@ -0,0 +1,482 @@
+	 */
+	public AndroidXBeeInterface(Context context, int baudRate) {
+		this(context, baudRate, null, null);
+	}
+
+	/**
+	 * Class constructor. Instantiates a new {@code AndroidXBeeInterface} object
+	 * with the given parameters.
+	 * 
+	 * @param context The Android context.
+	 * @param baudRate Device baud rate to use.
+	 * @param permissionListener Android USB permission listener to be notified 
+	 *                           when access to USB device is granted, may be
+	 *                           {@code null}.
+	 * 
+	 * @throws IllegalArgumentException if {@code baudRate < 1}.
+	 * @throws NullPointerException if {@code context == null}.
+	 * 
+	 * @see #AndroidXBeeInterface(Context, int)
+	 * @see #AndroidXBeeInterface(Context, int, UsbDevice)
+	 * @see #AndroidXBeeInterface(Context, int, UsbDevice, AndroidUSBPermissionListener)
+	 * @see AndroidUSBPermissionListener
+	 */
+	public AndroidXBeeInterface(Context context, int baudRate, AndroidUSBPermissionListener permissionListener) {
+		this(context, baudRate, null, permissionListener);
+	}
+	
+	/**
+	 * Class constructor. Instantiates a new {@code AndroidXBeeInterface} object
+	 * with the given parameters.
+	 * 
+	 * @param context The Android context.
+	 * @param baudRate Device baud rate to use.
+	 * @param usbDevice USB device to use, may be {@code null}.
+	 * @param permissionListener Android USB permission listener to be notified 
+	 *                           when access to USB device is granted, may be 
+	 *                           {@code null}.
+	 * 
+	 * @throws IllegalArgumentException if {@code baudRate < 1}.
+	 * @throws NullPointerException if {@code context == null}.
+	 * 
+	 * @see #AndroidXBeeInterface(Context, int)
+	 * @see #AndroidXBeeInterface(Context, int, AndroidUSBPermissionListener)
+	 * @see #AndroidXBeeInterface(Context, int, UsbDevice)
+	 * @see AndroidUSBPermissionListener
+	 * @see UsbDevice
+	 */
+	public AndroidXBeeInterface(Context context, int baudRate, UsbDevice usbDevice, AndroidUSBPermissionListener permissionListener) {
+		if (context == null)
+			throw new NullPointerException("Android context cannot be null.");
+		if (baudRate < 1)
+			throw new IllegalArgumentException("Baud rate must be greater than 0.");
+		
+		this.context = context;
+		this.baudRate = baudRate;
+		this.permissionListener = permissionListener;
+		this.usbDevice = usbDevice;
+		this.usbManager = (UsbManager)context.getSystemService(Context.USB_SERVICE);
+		this.logger = LoggerFactory.getLogger(AndroidXBeeInterface.class);
+	}
+
+	/**
+	 * Calculates a valid divisor for the given baud rate and base clock for
+	 * the FT232BM, FT2232C and FT232LR chips.
+	 * 
+	 * @param baud Desired baud rate. 
+	 * @param base The base clock.
+	 * 
+	 * @return The calculated divisor.
+	 */
+	private int calculateBaudRate(int baud, int base) {
+		int divisor;
+		divisor = (base / 16 / baud) | (((base / 2 / baud) & 4) != 0 ? 0x4000 // 0.5
```

Comment:
Could this be rewritten to not depend on chained ternary expressions?

Reply:
I'll change the method to:
```java
	private int calculateBaudRate(int baud, int base) {
		int divisor = (base / 16 / baud);
		if (((base / 2 / baud) & 4) != 0)
			divisor |= 0x4000; // 0.5
		else if (((base / 2 / baud) & 2) != 0)
			divisor |= 0x8000; // 0.25
		else if (((base / 2 / baud) & 1) != 0)
			divisor |= 0xC000; // 0.125
		else
			divisor |= 0;

		return divisor;
	}
```